Content.Server/SimpleStation14/Species/Shadowkin/Systems/ShadowkinSystem.cs

```diff
@@ -0,0 +1,157 @@
+
+            component.ForceSwapRate = _random.NextFloat(component.ForceSwapRateMin, component.ForceSwapRateMax);
+            component.TiredRate = _random.NextFloat(component.TiredRateMin, component.TiredRateMax);
+        }
+
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            var query = _entityManager.EntityQuery<ShadowkinComponent>();
+
+            // Update power level for all shadowkin
+            foreach (var component in query)
+            {
+                _powerSystem.TryUpdatePowerLevel(component.Owner, frameTime);
+                if (!component.Blackeye)
+                    _powerSystem.TryBlackeye(component.Owner);
+
+
+                // Sync client and server
+                component.DirtyAccumulator += frameTime;
+                if (component.DirtyAccumulator > component.DirtyAccumulatorRate)
+                {
+                    component.DirtyAccumulator = 0f;
+
+                    _powerSystem.UpdateAlert(component.Owner, true, component.PowerLevel);
+                    Dirty(component);
+                }
+
+
+                #region Random Stuff
+
+                #region ForceSwap
+                // Check if they're at max power
+                if (component.PowerLevel >= ShadowkinComponent.PowerThresholds[ShadowkinPowerThreshold.Max])
+                {
+                    // If so, start the timer
+                    component.ForceSwapAccumulator += frameTime;
+
+                    // If the timer is up, force swap
+                    if (component.ForceSwapAccumulator > component.ForceSwapRate)
+                    {
+                        // Reset timer
+                        component.ForceSwapAccumulator = 0f;
+                        // Random new timer
+                        component.ForceSwapRate = _random.NextFloat(component.ForceSwapRateMin, component.ForceSwapRateMax);
+
+                        // Add/Remove DarkSwapped component, which will handle the rest
+                        if (_entityManager.TryGetComponent<ShadowkinDarkSwappedComponent>(component.Owner, out var _))
+                        {
+                            RaiseNetworkEvent(new ShadowkinDarkSwappedEvent(component.Owner, false));
+                            _entityManager.RemoveComponent<ShadowkinDarkSwappedComponent>(component.Owner);
+                        }
+                        else
+                        {
+                            RaiseNetworkEvent(new ShadowkinDarkSwappedEvent(component.Owner, true));
+                            _entityManager.AddComponent<ShadowkinDarkSwappedComponent>(component.Owner);
+                        }
+                    }
+                }
+                else
+                {
+                    // Slowly regenerate if not max power
+                    component.ForceSwapAccumulator -= frameTime / 3f;
```

Comment:
I was writing comments locally on VScode and it was bein weird. I knew this might happen :P
I did not know that it would update comments before I pushed though, that's weird